eventauth.go

```diff
@@ -175,6 +175,61 @@ type AuthEvents interface {
 	ThirdPartyInvite(stateKey string) (*Event, error)
 }
 
+type stateKeyTuple struct {
+	Type     string
+	StateKey string
+}
+
+// AuthEventProviderMap is an implementation of AuthEventProvider backed by a map.
+type AuthEventProviderMap struct {
```

Comment:
Why not just call this `AuthEvents`? AuthEventProviderMap sounds like a map of AuthEventProviders to me.